not able to deploy cluster when addons not selected

```
*
* Phase 3: Deploying Addons
*
Run the addon manager? (phase3.run_addons) [Y/n/?] (NEW) n
#
# configuration written to .config
#
make: '.config' is up to date.
[container]:/opt/kubernetes-anywhere> make deploy
util/config_to_json .config > .config.json
make do WHAT=deploy-cluster
make[1]: Entering directory '/opt/kubernetes-anywhere'
( cd "phase1/$(jq -r '.phase1.cloud_provider' .config.json)"; ./do deploy-cluster )
RUNTIME ERROR: Field does not exist: phase3
		object <anonymous>
	std.jsonnet:28:53	function <anonymous>
	vSphere.jsonnet:(18:36)-(24:6)	thunk <config_metadata_template>
	vSphere.jsonnet:132:98-121	thunk <a>
	std.jsonnet:145:21	
	std.jsonnet:145:12-22	thunk <a>
	std.jsonnet:145:12-34	function <anonymous>
	std.jsonnet:145:12-34	function <anonymous>
	vSphere.jsonnet:132:98-132	thunk <b>
	std.jsonnet:148:27	thunk <vals>
	...
	vSphere.jsonnet:132:21-133	thunk <array_element>
	vSphere.jsonnet:(130:27)-(135:19)	object <anonymous>
	vSphere.jsonnet:(129:32)-(136:17)	object <anonymous>
	vSphere.jsonnet:(128:27)-(137:12)	thunk <array_element>
	vSphere.jsonnet:(128:26)-(141:13)	object <anonymous>
	vSphere.jsonnet:(121:17)-(142:9)	object <anonymous>
	std.jsonnet:(943:13)-(952:13)	object <anonymous>
	std.jsonnet:(943:13)-(952:13)	object <anonymous>
	all.jsonnet:3:49-79	object <anonymous>
	During manifestation	
Makefile:63: recipe for target 'do' failed
make[1]: *** [do] Error 1
make[1]: Leaving directory '/opt/kubernetes-anywhere'
Makefile:41: recipe for target 'deploy-cluster' failed
make: *** [deploy-cluster] Error 2
[container]:/opt/kubernetes-anywhere>
```

cc: @tusharnt @BaluDontu @abrarshivani

--- a/phase1/vsphere/Kconfig
+++ b/phase1/vsphere/Kconfig
@@ -55,28 +55,28 @@ endif
 
 if phase1.vSphere.placement = "cluster"
 	config phase1.vSphere.cluster
-        string "vspherecluster"
+        string "vsphere cluster name. Please make sure that all the hosts in the cluster are time-synchronized otherwise some of the nodes can remain in pending state for ever due to expired certificate"
         default ""
         help
 			vSphere cluster where kubernetes cluster will be created.
 endif
 
 config phase1.vSphere.useresourcepool
-	string "Do you want to use the resource pool created on the host or cluster? [yes, no]"
+	string "Do you want to use the existing resource pool on the host or cluster? [yes, no]"
 	default "no"
 	help
-		Deploy cluster on Resource Pool? Enter yes or no
+		Deploy cluster on existing Resource Pool? Enter yes or no
 
 if phase1.vSphere.useresourcepool = "yes"
 	config phase1.vSphere.resourcepool
-        string "Name of the Resource Pool. If Resource pool is enclosed within another Resource pool, specify pool hierarchy as ParentResourcePool/ChildResourcePool"
+        string "Name of the existing Resource Pool. If Resource pool is enclosed within another Resource pool, specify pool hierarchy as ParentResourcePool/ChildResourcePool"
         default ""
         help
-			Resource Pool where kubernetes cluster will be deployed.
+			Existing Resource Pool where kubernetes cluster will be deployed.
 endif
 
 config phase1.vSphere.vmfolderpath
-	string "VM Folder name or Path (e.g kubernetes, VMFolder1/dev-cluster, VMFolder1/Test Group1/test-cluster)"
+	string "VM Folder name or Path (e.g kubernetes, VMFolder1/dev-cluster, VMFolder1/Test Group1/test-cluster). Folder path will be created if not present"
 	default "kubernetes"
 	help
 		VM Folder Path where Node VMs should be placed.

--- a/phase1/vsphere/vSphere.jsonnet
+++ b/phase1/vsphere/vSphere.jsonnet
@@ -3,7 +3,7 @@ function(config)
   local cfg = config.phase1;
   local vms = std.makeArray(cfg.num_nodes + 1,function(node) node+1); 
   local master_dependency_list = ["vsphere_virtual_machine.kubevm%d" % vm for vm in vms];
-  local node_name_to_ip = [("${vsphere_virtual_machine.kubevm%d.network_interface.0.ipv4_address} %s"  % [vm, (if vm == 1 then "master" else "node%d" % (vm-1) )])  for vm in vms];
+  local node_name_to_ip = [("${vsphere_virtual_machine.kubevm%d.network_interface.0.ipv4_address} %s"  % [vm, (if vm == 1 then std.join("", [cfg.cluster_name, "-master"]) else std.join("", [cfg.cluster_name, "-node", vm-1]) )])  for vm in vms];
   local vm_username = "root";
   local vm_password = "kubernetes";
 
@@ -97,7 +97,7 @@ function(config)
       },
       vsphere_virtual_machine: {
         ["kubevm" + vm]: {
-            name: (if vm == 1 then "master" else ("node%d" % (vm-1))),
+            name: if vm == 1 then std.join("", [cfg.cluster_name, "-master"]) else std.join("", [cfg.cluster_name, "-node", vm-1]),
             vcpu: cfg.vSphere.vcpu,
             memory: cfg.vSphere.memory,
             enable_disk_uuid: true,
@@ -124,7 +124,7 @@ function(config)
         } for vm in vms
       },
       null_resource: {
-        master: {
+      [cfg.cluster_name + "-master"]: {
             depends_on: master_dependency_list,
             connection: {
               user: vm_username,
@@ -134,7 +134,7 @@ function(config)
             provisioner: [{
                 "remote-exec": {
                   inline: [
-                    "hostnamectl set-hostname %s" % "master",
+                    "hostnamectl set-hostname %s" % std.join("", [cfg.cluster_name, "-master"]),
                     "mkdir -p /etc/kubernetes/; echo '%s' > /etc/kubernetes/k8s_config.json " % (config_metadata_template % "master"),                    
                     "echo '%s' >  /etc/kubernetes/vsphere.conf" % "${data.template_file.cloudprovider.rendered}",            
                     "echo '%s' > /etc/configure-vm.sh; bash /etc/configure-vm.sh" % "${data.template_file.configure_master.rendered}",
@@ -146,7 +146,7 @@ function(config)
             },
            }],
         },} + {
-        ["node" + vm]: {
+        [cfg.cluster_name + "-node" + vm]: {
             depends_on: ["vsphere_virtual_machine.kubevm1","vsphere_virtual_machine.kubevm%d" % vm],
             connection: {
               user: vm_username,
@@ -156,7 +156,7 @@ function(config)
             provisioner: [{
                 "remote-exec": {
                   inline: [
-                    "hostnamectl set-hostname %s" % ("node" + (vm-1)),
+                    "hostnamectl set-hostname %s" % std.join("", [cfg.cluster_name, "-node", vm-1]),
                     "mkdir -p /etc/kubernetes/; echo '%s' > /etc/kubernetes/k8s_config.json " % (config_metadata_template % "node"),                    
                     "echo '%s' > /etc/configure-vm.sh; bash /etc/configure-vm.sh" % "${data.template_file.configure_node.rendered}",
                     "echo '%s' >  /etc/kubernetes/vsphere.conf" % "${data.template_file.cloudprovider.rendered}",            

--- a/phase3/all.jsonnet
+++ b/phase3/all.jsonnet
@@ -1,5 +1,5 @@
 function(cfg)
-  local if_enabled(addon, manifest) = if std.objectHas(cfg.phase3, addon) && cfg.phase3[addon] then manifest else {};
+  local if_enabled(addon, manifest) = if std.objectHas(cfg, "phase3") && std.objectHas(cfg.phase3, addon) && cfg.phase3[addon] then manifest else {};
   local join(arr) = std.foldl(function(a, b) a + b, arr, {});
   if_enabled("run_addons",
              join([

--- a/phase3/do
+++ b/phase3/do
@@ -18,7 +18,7 @@ gen() {
 deploy() {
 	gen
 	mkdir -p /tmp/kubectl/
-	HOME=/tmp/kubectl kubectl apply -f ./${TMP_DIR}
+	HOME=/tmp/kubectl kubectl apply -f ${TMP_DIR}
 }
 
 

--- a/util/validate
+++ b/util/validate
@@ -13,7 +13,7 @@ num_nodes="$(( $(cat ${CONFIG_FILE} | jq -r '.phase1.num_nodes') + 1 ))"
 while true; do
   total_time=$(( ${total_time} + ${wait_duration} ))
   hcount=$(kubectl get nodes 2>/dev/null | grep 'Ready' | wc -l) || true
-  echo "Validation: Expected ${num_nodes} healthy nodes; found ${hcount}. (${total_time}s elapsed)"
+  echo "Validation: Expected ${num_nodes} (workers + master) healthy nodes; found ${hcount}. (${total_time}s elapsed)"
 
   [[ "${hcount}" -ge "${num_nodes}" ]] && echo "Validation: Success!" && exit
 

